pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go

```diff
@@ -224,31 +227,48 @@ func doesSliceContainStr(slice []string, str string) (isFound bool) {
 	return isFound
 }
 
-func validateCloneSourceVM(client kubecli.KubevirtClient, name, namespace string, sourceField *k8sfield.Path) []metav1.StatusCause {
-	vm, err := client.VirtualMachine(namespace).Get(name, &metav1.GetOptions{})
-	if errors.IsNotFound(err) {
+func validateCloneSourceExists(clientGetErr error, sourceField *k8sfield.Path, kind, name, namespace string) []metav1.StatusCause {
+	if errors.IsNotFound(clientGetErr) {
 		return []metav1.StatusCause{
 			{
 				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: fmt.Sprintf("Virtual Machine %s does not exist in namespace %s", name, namespace),
+				Message: fmt.Sprintf("%s %s does not exist in namespace %s", kind, name, namespace),
 				Field:   sourceField.String(),
 			},
 		}
-	} else if err != nil {
+	} else if clientGetErr != nil {
 		return []metav1.StatusCause{
 			{
 				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: fmt.Sprintf("error occurred while trying to get source VM: %v", err),
+				Message: fmt.Sprintf("error occurred while trying to get source %s: %v", kind, clientGetErr),
 				Field:   sourceField.String(),
 			},
 		}
 	}
 
+	return nil
+}
+
+func validateCloneSourceVM(client kubecli.KubevirtClient, name, namespace string, sourceField *k8sfield.Path) []metav1.StatusCause {
+	vm, err := client.VirtualMachine(namespace).Get(name, &metav1.GetOptions{})
+	causes := validateCloneSourceExists(err, sourceField, "VirtualMachine", name, namespace)
+
+	if causes != nil {
+		return causes
+	}
+
 	// currently, cloning leverages vm snapshot/restore to copy volumes
 	// snapshot/restore requires that volumes support CSI snapshots
 	// this limitation should be removed eventually
 	// probably by leveraging CDI cloning
-	return validateCloneVolumeSnapshotSupportVM(vm, sourceField)
+	causes = append(causes, validateCloneVolumeSnapshotSupportVM(vm, sourceField)...)
+
+	return causes
+}
+
+func validateCloneSourceSnapshot(client kubecli.KubevirtClient, name, namespace string, sourceField *k8sfield.Path) []metav1.StatusCause {
+	_, err := client.VirtualMachineSnapshot(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	return validateCloneSourceExists(err, sourceField, "VirtualMachineSnapshot", name, namespace)
```

Comment:
have to do some validation here similar to `validateCloneVolumeSnapshotSupportVM`.  Must inspect the VMSnapshotContent VM yaml to confirm that all pvc/dv volumes have VolumeBackups.  Otherwise, the vm yaml will refer to a pvc that may not exist.  Or if it exists, it will end up being shared by all clones.

Reply:
Thanks!
Main related changes are in https://github.com/kubevirt/kubevirt/pull/8458/commits/52ba8b65800e611ecf2675f62a3edb53d8793acb